php/Blockonomics.php

```diff
@@ -238,4 +292,267 @@ public function testSetup()
+                $network_confirmations=get_option("blockonomics_network_confirmation",2);
+                if ($status == 0 && time() > $timestamp + $time_period) {
+                    $minutes = (time() - $timestamp)/60;
+                    $wc_order->add_order_note(__("Warning: Payment arrived after $minutes minutes. Received". $order['crypto'] ."may not match current ". $order['crypto'] ." price", 'blockonomics-bitcoin-payments'));
+                }
+                elseif ($status >= $network_confirmations && !metadata_exists('post',$wc_order->get_id(),'paid_'. $order['crypto'] .'_amount') )  {
+                    update_post_meta($wc_order->get_id(), 'paid_'. $order['crypto'] .'_amount', $value/1.0e8);
+                    if ($order['satoshi'] > $value) {
+                        //Check underpayment slack
+                        $underpayment_slack = get_option("blockonomics_underpayment_slack", 0)/100 * $order['satoshi'];
+                        if ($order['satoshi'] - $underpayment_slack > $value) {
+                            $status = -2; //Payment error , amount not matching
+                            $wc_order->update_status('failed', __('Paid '. $order['crypto'] .' amount less than expected.', 'blockonomics-bitcoin-payments'));
+                        }else{
+                            $wc_order->add_order_note(__('Payment completed', 'blockonomics-bitcoin-payments'));
+                            $wc_order->payment_complete($order['txid']);
+                        }
+                    }
+                    else{
+                        if ($order['satoshi'] < $value) {
+                            $wc_order->add_order_note(__('Overpayment of '. $order['crypto'] .' amount', 'blockonomics-bitcoin-payments'));
+                        }
+                        $wc_order->add_order_note(__('Payment completed', 'blockonomics-bitcoin-payments'));
+                        $wc_order->payment_complete($order['txid']);
+                    }
+                    // Keep track of funds in temp wallet
+                    if(get_option('blockonomics_temp_api_key') && !get_option("blockonomics_api_key")) {
+                        $current_temp_amount = get_option('blockonomics_temp_withdraw_amount');
+                        $new_temp_amount = $current_temp_amount + $value;
+                        update_option('blockonomics_temp_withdraw_amount', $new_temp_amount);
+                    }
+                }
+                $order['txid'] = $txid;
+                $order['status'] = $status;
+                $orders[$order_id][$addr] = $order;
+                if ($existing_status == -1) {
+                    update_post_meta($wc_order->get_id(), 'blockonomics_'. $order['crypto'] .'_txid', $order['txid']);
+                    update_post_meta($wc_order->get_id(), 'expected_'. $order['crypto'] .'_amount', $order['satoshi']/1.0e8);
+                }
```

Comment:
This extra condition is necessary now that we are checking for paid_crypto metadata

Reply:
Refactored in 62c1ec2